packages/react/src/products/hooks/__tests__/useRecentlyViewedProducts.test.tsx

```diff
@@ -0,0 +1,262 @@
+              mobilePageSize: 9,
+            },
+          },
+        },
+      },
+    };
+
+    const { result } = renderHook(
+      () =>
+        useRecentlyViewedProduct({
+          excludeProductId: mockProductId,
+        }),
+      {
+        wrapper: withStore(mockState),
+      },
+    );
+
+    expect(result.current).toStrictEqual({
+      error: undefined,
+      isFetched: true,
+      isLoading: false,
+      data: {
+        number: 1,
+        totalPages: 1,
+        totalItems: 2,
+        products: expectedProductsWithBrand,
+      },
+      actions: {
+        fetch: expect.any(Function),
+        remove: expect.any(Function),
+        save: expect.any(Function),
+      },
+    });
+  });
+
+  it('should return error state', () => {
+    const mockError = { message: 'This is an error message' };
+
+    const errorMockRecentlyViewedState = {
+      entities: {},
+      products: {
+        lists: {
+          error: {},
+          isHydrated: {},
+          isLoading: {},
+          hash: null,
+        },
+        recentlyViewed: {
+          error: mockError,
+          isLoading: false,
+          result: null,
+        },
+      },
+    };
+
+    const { result } = renderHook(
+      () => useRecentlyViewedProduct({ excludeProductId: mockProductId }),
+      {
+        wrapper: withStore(errorMockRecentlyViewedState),
+      },
+    );
+
+    expect(result.current).toStrictEqual({
+      error: mockError,
+      isFetched: true,
+      isLoading: false,
+      data: {
+        products: undefined,
+      },
+      actions: {
+        fetch: expect.any(Function),
+        remove: expect.any(Function),
+        save: expect.any(Function),
+      },
+    });
+  });
+
+  it('should return loading state', () => {
+    const loadingMockRecentlyViewedState = {
+      entities: {},
+      products: {
+        lists: {
+          error: {},
+          isHydrated: {},
+          isLoading: {},
+          hash: null,
+        },
+        recentlyViewed: {
+          error: null,
+          isLoading: true,
+          result: null,
+        },
+      },
+    };
+
+    const { result } = renderHook(
+      () => useRecentlyViewedProduct({ excludeProductId: mockProductId }),
+      {
+        wrapper: withStore(loadingMockRecentlyViewedState),
+      },
+    );
+
+    expect(result.current).toStrictEqual({
+      error: undefined,
+      isFetched: false,
+      isLoading: true,
+      data: {
+        products: undefined,
+      },
+      actions: {
+        fetch: expect.any(Function),
+        remove: expect.any(Function),
+        save: expect.any(Function),
+      },
+    });
+  });
+
+  describe('options', () => {
+    it('should not fetch data if `enableAutoFetch` option is false', () => {
+      renderHook(
+        () =>
+          useRecentlyViewedProduct({
+            excludeProductId: mockProductId,
+            enableAutoFetch: false,
+          }),
+        {
+          wrapper: withStore(mockProductsState),
```

Comment:
This state already prevents the refetch because it already has data. Please, use another state which satisfies the condition for the useEffect so it can be properly tested.